src/api/note_service.py

```diff
@@ -1,12 +1,42 @@
+from collections.abc import Sequence
+from ..db import notes as NotesDB
 from ..models.note import Note
-from ..db.notes import get_notes_for_user, create_note
 
 
-def get_notes(user_id: int) -> list[Note]:
-    return get_notes_for_user(user_id)
+def get_notes(author_id: int) -> Sequence[Note]:
+    return NotesDB.get_notes_for_user(author_id)
 
 
-def post_note(
+def get_note_by_id(author_id: int, note_id: int) -> Note | None:
+    note = NotesDB.get_note_by_id(note_id)
+    if note is None:
+        return None
+
+    print(type(note_id))
+    if note.is_public or note.author_id == author_id:
+        return note
+
+    return None
+
+    print(note.author_id)
+    return note if note.is_public or note.author_id == author_id else None
```

Comment:
These last two lines are unreachable because of the return statement above

Reply:
No idea how this happened lol, let me fix it